### PR TITLE
add explicit Dbghelp dependency on Windows for compilers other than MSVC

### PR DESCRIPTION
**Pull request by Optiligence**

### CODE REVIEW

**Comment by andschwa** [comment]
I have no objection to this. It's redundant when using MSVC, but shouldn't cause a problem with it, and adds better CMake integration with non-MSVC compilers.

👍

File: CMakeLists.txt [hunk 1]
```diff
@@ -410,6 +410,8 @@ add_library (glog
 
 set_target_properties (glog PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
+target_link_libraries(${PROJECT_NAME} PUBLIC $<$<PLATFORM_ID:Windows>:Dbghelp>)
```
Comment (andschwa) [inline CMakeLists.txt]: `s/${PROJECT_NAME}/glog` for consistency with the other uses of the target.

Comment (Optiligence) [inline CMakeLists.txt]: done

**Comment by sergiud** [comment]
I've addressed this in PR #208 already.

**Comment by andschwa** [comment]
Oh yeah, I forgot about that. Thanks @sergiud.

@Optiligence you can close this, it's getting fixed. Thanks for your contribution.

**Comment by Optiligence** [comment]
I like my line more^^ (`DbgHelp` should always be available anyway)
Nice to hear it’s getting fixed.

**Comment by sergiud** [comment]
You patch is incomplete. The platform ID does not imply the existence of `Dbghelp`. You need to actually check whether the library exists.

**Comment by Optiligence** [comment]
`Dbghelp` always exists, doesn’t it?

**Comment by Optiligence** [comment]
For the functions being used it’s shipped with Windows since XP, i doubt further support is needed.

**Comment by sergiud** [comment]
You're missing the point. Your check is too simplistic to cover all possibilities, e.g., in cross-compilation scenarios where dbghelp might actually not be available for whatever reasons.

**Comment by Optiligence** [comment]
In general, I agree, but in this case I don’t. 
Similarly nobody is going to search for `user32` too. 
If `Dbghelp` isn’t available the code won’t compile anyway, so what’s the point.

**Pull request closed without merging**